src/runtime_src/core/pcie/tools/xbutil/xbutil.h

```diff
@@ -404,6 +407,46 @@ class device {
         return 0;
     }
 
+    /* new KDS which supported CU subdevice */
+    int parseCUSubdevStat() const
+    {
+        using tokenizer = boost::tokenizer< boost::char_separator<char> >;
+        boost::char_separator<char> sep(" ");
+        std::vector<std::string> custat;
+        std::string errmsg;
+
+        pcidev::get_dev(m_idx)->sysfs_get("", "kds_custat_raw", errmsg, custat);
+        for (auto& line : custat) {
```

Comment:
I am assuming you are parsing this formatted string:
` "%d %s:%s 0x%llx 0x%x %llu"`
If so, you might wish to help the next developer and add a comment describing the string that is being parsed.

Reply:
fixed